api/get.js

```diff
@@ -0,0 +1,179 @@
+                        Q.nfcall(fs.readFile, result.replace(/\.req$/, '.stats'), { encoding: 'utf8' }),
+                        Q.nfcall(fs.readFile, result.replace(/\.req$/, '.res'), { encoding: 'utf8' })
+                    ])
+                    .then(function(files) {
+                        var stats = JSON.parse(files[0].value);
+                        res.status(stats.status);
+                        _.each(stats.headers, function(value, key) {
+                            res.setHeader(key, value);
+                        });
+                        res.write(files[1].value);
+                        res.end();
+                        resolve(result);
+                    })
+                    .fail(reject);
+                });
+            });
+    },
+    
+    request: function(url, fileList, req, res, next) {
+        return Q.Promise(function(resolve, reject) {
+            console.log('file not found, requesting from', url + req.url);
+            request(url + req.url, function(error, response, body) {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                if (response.statusCode === 200) {
+                    this.save(req.params, response, body, fileList.length)
+                        .then(function() {
+                            res.status(response.statusCode);
+                            _.each(response.headers, function(value, key) {
+                                res.setHeader(key, value);
+                            });
+                            res.write(body);
+                            res.end();
+                            resolve();
+                        })
+                        .fail(reject);
+                    return;
+                }
+                reject(new Error(body));
+            }.bind(this));
+        }.bind(this));
+    },
+    
+    save: function(params, res, body, count) {
+        console.log('saving response content', count, params);
+        return Q.allSettled([
+            Q.Promise(function(resolve, reject) {
+                console.log('save to', path.join(this.folder, 'file_' + count + '.req'));
+                fs.writeFile(path.join(this.folder, 'file_' + count + '.req'), JSON.stringify(params, null, 2), function(err) {
+                    if (!err) {
+                        console.log('file_' + count + '.req saved!');
+                        resolve();
+                    } else {
+                        reject(err);
+                    }
+                });
+            }.bind(this)),
+            Q.Promise(function(resolve, reject) {
+                console.log('save to', path.join(this.folder, 'file_' + count + '.stats'));
+                fs.writeFile(path.join(this.folder, 'file_' + count + '.stats'), JSON.stringify({
+                    status: res.statusCode,
+                    headers: res.headers
+                }, null, 2), function(err) {
```

Comment:
",null, 2)" - this is the 2nd time I see it. Let's not have magic numbers and null flying around. Can we create a common object for this?